WordPress/gutenberg
# Date: add relative time translations for moment.js

Resolves https://github.com/WordPress/gutenberg/issues/53888

## What?
Overwrites `wp-date` script string to include new `relativeTime` translations.

The inline script is generated in Core here: https://github.com/WordPress/wordpress-develop/blob/bf00a673a5f1d7f4707a6f914e1408526a8dc06a/src/wp-includes/script-loader.php#L430 and will need to be synced.

See: https://momentjscom.readthedocs.io/en/latest/moment/07-customization/07-relative-time/

## Why?
The [settings strings](https://github.com/WordPress/gutenberg/blob/eed863a1aacecf166b43eaa90077ccd0f287ad41/packages/date/src/index.js#L111-L111) were never synced to Core. 

Now that we're using `relativeTime` in [humanTimeDiff](https://github.com/WordPress/gutenberg/blob/eed863a1aacecf166b43eaa90077ccd0f287ad41/packages/date/src/index.js#L592C8-L592C44), we need the translations.

Unfortunately, `%d minutes` and others have not yet been translated.

## How?

Hooking into `wp_default_scripts` and overwriting the settings of the registered `wp-date` script.

The other option is to re-add the inline script, which will also work, but it will output 2 x inline scripts, e.g., 

```html
<script id="wp-date-js-after">
wp.date.setSettings( ...oldSettings );
wp.date.setSettings( ...newSettings );
</script>
```

## Testing Instructions
Change your editor/site to another language via Settings > General
Since the relevant translations have not been added, you can check that the `wp.date` settings have been updated by heading to `/wp-admin/site-editor.php` and, in the browser console, running `wp.date.getSettings()`

The `relative` object should match the requirements of https://momentjscom.readthedocs.io/en/latest/moment/07-customization/07-relative-time/

```js
{
    "l10n": {
        "locale": "de_DE",
        "months": [
            "Januar",
            "Februar",
            "März",
            "April",
            "Mai",
            "Juni",
            "Juli",
            "August",
            "September",
            "Oktober",
            "November",
            "Dezember"
        ],
        "monthsShort": [
            "Jan",
            "Feb",
            "Mrz",
            "Apr",
            "Mai",
            "Jun",
            "Jul",
            "Aug",
            "Sep",
            "Okt",
            "Nov",
            "Dez"
        ],
        "weekdays": [
            "Sonntag",
            "Montag",
            "Dienstag",
            "Mittwoch",
            "Donnerstag",
            "Freitag",
            "Samstag"
        ],
        "weekdaysShort": [
            "So",
            "Mo",
            "Di",
            "Mi",
            "Do",
            "Fr",
            "Sa"
        ],
        "meridiem": {
            "am": "am",
            "pm": "pm",
            "AM": "AM",
            "PM": "PM"
        },
        "relative": {
            "future": "%s ab jetzt",
            "past": "vor %s",
            "s": "a second",
            "ss": "%d seconds",
            "m": "a minute",
            "mm": "%d minutes",
            "h": "an hour",
            "hh": "%d hours",
            "d": "a day",
            "dd": "%d days",
            "M": "a month",
            "MM": "%d months",
            "y": "a year",
            "yy": "%d years"
        },
        "startOfWeek": 1
    },
    "formats": {
        "time": "g:i a",
        "date": "F j, Y",
        "datetime": "j. F Y G:i",
        "datetimeAbbreviated": "j. M. Y G:i"
    },
    "timezone": {
        "offset": 0,
        "string": "",
        "abbr": ""
    }
}
```
